pkg/restore/restore.go

```diff
@@ -525,16 +525,26 @@ func (ctx *restoreContext) execute() (results.Result, results.Result) {
 		var w, e results.Result
 		// Restore this resource, the update channel is set to nil, to avoid misleading value of "totalItems"
 		// more details see #5990
-		processedItems, w, e = ctx.processSelectedResource(
+		processedItems, createdItems, w, e = ctx.processSelectedResource(
 			selectedResource,
 			totalItems,
 			processedItems,
+			createdItems,
 			existingNamespaces,
 			nil,
 		)
 		warnings.Merge(&w)
 		errs.Merge(&e)
 	}
+	// If we just restored custom resource definitions (CRDs), refresh
+	// discovery because the restored CRDs may have created new APIs that
+	// didn't previously exist in the cluster, and we want to be able to
+	// resolve & restore instances of them in subsequent loop iterations.
+	if createdItems > 0 {
```

Comment:
How about refreshing the discovery helper no matter whether new items are created or not? This is much easier.

Reply:
yes, it is easier, but if no crds are created, refresh is meaningless, and it takes time (~9s in my env)